grammar.js

```diff
@@ -160,32 +166,53 @@ module.exports = grammar({
     /// Parameters
 
     parameter_parens: ($) =>
-      seq(BRACK().open_paren, repeat($.parameter), BRACK().close_paren),
+      seq(
+        BRACK().open_paren,
+        repeat($.parameter),
+        repeat($._newline),
+        BRACK().close_paren,
+      ),
 
     parameter_bracks: ($) =>
-      seq(BRACK().open_brack, repeat($.parameter), BRACK().close_brack),
+      seq(
+        BRACK().open_brack,
+        repeat($.parameter),
+        repeat($._newline),
+        BRACK().close_brack,
+      ),
 
-    parameter_pipes: ($) => seq(PUNC().pipe, repeat($.parameter), PUNC().pipe),
+    parameter_pipes: ($) =>
+      seq(
+        repeat($._newline),
+        PUNC().pipe,
+        repeat($.parameter),
+        repeat($._newline),
+        PUNC().pipe,
+      ),
 
     parameter: ($) =>
-      seq(
-        choice(
-          $._param_name,
-          field("param_long_flag", $.param_long_flag),
-          field(
-            "param_long_flag",
-            seq($.param_long_flag, field("flag_capsule", $.flag_capsule)),
-          ),
-        ),
-        optional(
+      prec.right(
+        seq(
+          repeat($._newline),
```

Comment:
Since `$.parameter` is repeated, I feel it more clear that it only allows edge newlines on one side (here on the beginning). Shared newline across parameters probably wont cause any conflict though.

Reply:
I hope that tests wouldn't pass on conflict, I was just wondering. Consider it more a curious question than a request for change :)

Reply:
Actually, it's a good point, I probably need to use a helper function for those kind of repeated patterns, somewhat like list_entries.